### PR TITLE
Fix node crashing when multiple TXs for same account are in slot

### PR DESCRIPTION
**Pull request by CodeSandwich**
Fixes https://github.com/input-output-hk/jormungandr/issues/586

### CODE REVIEW

File: jormungandr/src/blockchain/process.rs [hunk 1]
```diff
@@ -33,20 +33,18 @@ pub fn handle_input(
         BlockMsg::LeadershipExpectEndOfEpoch(epoch) => {
             let blockchain = blockchain.lock_read();
             chain::handle_end_of_epoch_event(&blockchain, epoch)
-                .map_err(|e| crit!(logger, "end of epoch processing failed: {:?}", e))?;
+                .map_err(|e| crit!(logger, "end of epoch processing failed: {:?}", e));
```
Comment (NicolasDP) [inline jormungandr/src/blockchain/process.rs]: not keen on silently ignoring the error though. This is critical and if the end of epoch processing fails : it means we will not be able to start processing the epoch, the loop is broken and actions need to be performed to address the issue.

Comment (CodeSandwich) [reply jormungandr/src/blockchain/process.rs]: Won't the node just skip its slot leadership? It's not going to be a leader of the next one, so other node will produce it and failing node at some point will get ultimate ledger truth from the network. At least that's what I thought, could be wrong though.

Comment (NicolasDP) [reply jormungandr/src/blockchain/process.rs]: This part is not about a _slot_.

Comment (NicolasDP) [reply jormungandr/src/blockchain/process.rs]: have a look at the leadership doc https://github.com/input-output-hk/jormungandr/blob/e8f0cc2684a9bd5e7d8758fc39299280bd24e111/jormungandr/src/leadership/mod.rs#L3-L85

Comment (NicolasDP) [reply jormungandr/src/blockchain/process.rs]: you can also quickly generate the dev doc with `cargo doc --open --no-deps` and explore code documentation for a better experience.

Comment (mzabaluev) [reply jormungandr/src/blockchain/process.rs]: I believe an error here can be only a serious internal problem that should cause the whole task to call it quits.

Comment (NicolasDP) [reply jormungandr/src/blockchain/process.rs]: for now at least. we will need to see what else we can do. But not now.

Comment (CodeSandwich) [reply jormungandr/src/blockchain/process.rs]: Ah, right, this is exact line is not about being a leader. I've restored the task death in that case.

There's also similar case for `BlockMsg::LeadershipBlock`, where `chain::handle_block` failure stopped being a task killer. This one IS about being a leader, is that ok? https://github.com/input-output-hk/jormungandr/pull/619/files#diff-d93e6a17addd0650d4e7ded097ca6bb0L41

**Pull request merged**